pydarn/plotting/rtp.py

```diff
@@ -26,22 +29,602 @@ class RTP(matplotlib.pyplot):
+                            # TODO: refactor and clean up this code
+                            elif cls.__filter_data_check(dmap_record,
+                                                         settings, j):
+                                # check if boundaries have been set
+                                if settings["boundary"]:
+                                    # see if data within boundaries
+                                    if z_min < dmap_record[parameter][j] and\
+                                       z_max > dmap_record[parameter][j]:
+                                        z[i][dmap_record['slist'][j]] = \
+                                                dmap_record[parameter][j]
+                                else:
+                                    z[i][dmap_record['slist'][j]] = \
+                                            dmap_record[parameter][j]
+                                    # calculate min and max value
+                                    if z[i][dmap_record['slist'][j]] < z_min or\
+                                       z_min is None:
+                                        z_min = z[i][dmap_record['slist'][j]]
+                                    if z[i][dmap_record['slist'][j]] > z_max:
+                                        z_max = z[i][dmap_record['slist'][j]]
+                    # a KeyError may be thrown because slist is not created
+                    # due to bad quality data.
+                    except KeyError:
+                        continue
+
+        # Check if there is any data to plot
+        if np.all(np.isnan(z)):
+            raise rtp_exceptions.RTPNoDataFoundError(parameter, beam_num,
+                                                     start_time, end_time)
+        time_axis, elev_axis = np.meshgrid(x, y)
+        z_data = np.ma.masked_where(np.isnan(z.T), z.T)
+
+        norm = colors.Normalize(z_min, z_max)
+
+        cmap = cm.get_cmap(settings['color_map'])
+        # set ground scatter to grey
+        cmap.set_under('grey', 1.0)
+
+        # plot!
+        im = ax.pcolormesh(time_axis, elev_axis, z_data, lw=0.01,
+                           cmap=cmap, norm=norm)
+        # setup some standard axis information
+        ax.set_xlim([start_time, end_time])
+        ax.xaxis.set_major_formatter(dates.DateFormatter(settings['date_fmt']))
+        ax.yaxis.set_ticks(np.arange(0, y_max+1, 15))
+        ax.xaxis.set_minor_locator(dates.HourLocator())
+        ax.yaxis.set_minor_locator(ticker.MultipleLocator(5))
+        # so the plots gets to the ends
+        ax.margins(0)
+
+        # create color bar if True
+        cb = None
+        if settings['color_bar']:
+            cb = ax.figure.colorbar(im, ax=ax)
+            cb.set_label(settings['color_bar_label'])
+
+        return im, cb, cmap
+
+    @classmethod
+    def plot_time_series(cls, dmap_data: List[dict], *args,
+                         parameter: str = 'frequency', beam_num: int = 0,
```

Comment:
All other `str=` here don't have whitespace.  Recommend being consistent.

Reply:
when you use type hinting the PEP8 linter I use (Flake8) does not recommend spaces around the equal sign. However, for non-typed hinted variables you don't put whitespace around the equal sign. 

But, I ran flake8 on this script and did find some whitespace problems so I fixed those :)

Reply:
I guess I am being consistent with Flake8 suggestions which is a PEP8 linter.